prow/io/opener.go

```diff
@@ -48,13 +55,26 @@ type (
 	WriteCloser = io.WriteCloser
 )
 
+type Attributes struct {
+	// ContentEncoding specifies the encoding used for the blob's content, if any.
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
+	ContentEncoding string
+	// Size is the size of the blob's content in bytes.
+	Size int64
+}
+
 // Opener has methods to read and write paths
 type Opener interface {
 	Reader(ctx context.Context, path string) (ReadCloser, error)
+	RangeReader(ctx context.Context, path string, offset, length int64) (io.ReadCloser, error)
 	Writer(ctx context.Context, path string, opts ...WriterOptions) (WriteCloser, error)
+	Attributes(ctx context.Context, path string) (Attributes, error)
+	SignedURL(ctx context.Context, path string, opts SignedURLOptions) (string, error)
```

Comment:
It is an interaction with the underlying storage provider that needs a similar set of information (what provider is it, credentials). If we don't do this, we will have to pass around both an `io.Opener` and cloud provider credentials in Deck.

You might now argue that this leaks through implementation details since it does not apply to block storage, but my opinion is that in the context of Prow, object storage is the first class citizen and block storage is more of a fallback.